### PR TITLE
VZ-8907 added diagnosis  of cert issues to vz analyze

### PR DESCRIPTION
**Pull request by absBitm**
- added diagnosis of cert issues to vz analyze during vz install

### CODE REVIEW

File: tools/vz/pkg/analysis/internal/util/cluster/install.go [hunk 1]
```diff
@@ -7,6 +7,7 @@ package cluster
 import (
 	encjson "encoding/json"
 	"fmt"
+	"github.com/verrazzano/verrazzano/tests/e2e/pkg"
```
Comment (agv426) [inline tools/vz/pkg/analysis/internal/util/cluster/install.go:10]: Why are you importing a test package into the CLI? Is there something that should be in the products common code?

Comment (absBitm) [reply tools/vz/pkg/analysis/internal/util/cluster/install.go:10]: I could not find any such method other way I can paste the same method inside tools/vz/pkg/helpers/vzhelper.go

File: tools/vz/pkg/analysis/internal/util/cluster/install.go [hunk 2]
```diff
@@ -132,6 +144,46 @@ func checkIfLoadJobFailed(vpoLog string, errorLogs []files.LogMessage, infoLogs
 	}
 }
 
+func analyzeComponentCertIssue(log *zap.SugaredLogger, clusterRoot string, issueReporter *report.IssueReporter, allMessages []files.LogMessage, vpoLog string) {
+	certs, err := pkg.GetCertificateList("")
+	if err != nil {
+		return
+	}
+	issueVisited := make(map[string]bool)
+	certCompMap := map[string]string{
```
Comment (bhatpmk) [inline tools/vz/pkg/analysis/internal/util/cluster/install.go:153]: Maintaining the list of certificates in the analysis tool will make the maintenance difficult. Ideally we need this at common location, shared with platform operator. The intention is to get the new certificate info to the CLI automatically.

File: tools/vz/pkg/analysis/internal/util/cluster/install.go [hunk 3]
```diff
@@ -62,6 +64,16 @@ const (
 	nginxIngressControllerFailed = "nginxIngressControllerFailed"
 	noIPForIngressController     = "noIPForIngressController"
 	errorSettingRancherToken     = "errorSettingRancherToken"
+	grafanaComponent             = "grafana"
```
Comment (bhatpmk) [inline tools/vz/pkg/analysis/internal/util/cluster/install.go:67]: Please avoid defining constants for the components again in the CLI. You will get some of them from pkg/constants/constants.go, the remaining can be added to constants.go and hard-coded values can be removed from the respective _component.go

File: tools/vz/pkg/analysis/internal/util/cluster/install.go (hunk 1 above)
Comment (bhatpmk) [reply tools/vz/pkg/analysis/internal/util/cluster/install.go:10]: As Tony mentioned, please avoid using the function defined in tests/e2e/pkg.

File: tools/vz/pkg/analysis/internal/util/cluster/install.go (hunk 2 above)
Comment (absBitm) [reply tools/vz/pkg/analysis/internal/util/cluster/install.go:152]: ok

File: tools/vz/pkg/analysis/internal/util/cluster/install.go (hunk 1 above)
Comment (absBitm) [reply tools/vz/pkg/analysis/internal/util/cluster/install.go]: yes its under tools/vz/pkg/helpers/vzhelper.go now

File: tools/vz/pkg/analysis/internal/util/cluster/install.go (hunk 3 above)
Comment (absBitm) [reply tools/vz/pkg/analysis/internal/util/cluster/install.go:66]: I did check, it was not there
but I will move this const to pkg/constants/constants.go

File: tools/vz/pkg/analysis/internal/util/cluster/install.go [hunk 4]
```diff
@@ -132,6 +134,46 @@ func checkIfLoadJobFailed(vpoLog string, errorLogs []files.LogMessage, infoLogs
...
+		pkgConstants.Verrazzano + "-tls":                      pkgConstants.Verrazzano,
+	}
+
+	files := StringSlice{clusterRoot + "/" + verrazzanoResource, vpoLog}
+	if helpers.GetIsLiveCluster() {
+		files[0] = report.GetRelatedVZResourceMessage()
+		files[1] = report.GetRelatedLogFromPodMessage(vpoLog)
+	}
+
+	for _, msg := range allMessages {
```
Comment (bhatpmk) [inline tools/vz/pkg/analysis/internal/util/cluster/install.go:163]: I think the tool can report an issue whenever one or more well known Certificates are not ready, without going through all the lines of VPO log.

Comment (absBitm) [reply tools/vz/pkg/analysis/internal/util/cluster/install.go:163]: we need concrete issue and its cause which we fetch from vpo logs now

File: tools/vz/pkg/analysis/internal/util/cluster/install.go [hunk 5]
```diff
@@ -132,6 +134,46 @@ func checkIfLoadJobFailed(vpoLog string, errorLogs []files.LogMessage, infoLogs
 	}
 }
 
+func analyzeComponentCertIssue(log *zap.SugaredLogger, clusterRoot string, issueReporter *report.IssueReporter, allMessages []files.LogMessage, vpoLog string) {
```
Comment (bhatpmk) [inline tools/vz/pkg/analysis/internal/util/cluster/install.go:137]: Please add a comment to describe the function

File: tools/vz/pkg/analysis/internal/util/cluster/install_test.go [hunk 6]
```diff
@@ -29,4 +29,5 @@ func TestAnalyzeVerrazzanoResource(t *testing.T) {
 	assert.NoError(t, AnalyzeVerrazzanoResource(logger, "../../../test/cluster/istio-ingress-ip-not-found/cluster-snapshot", &issueReporter))
 	assert.NoError(t, AnalyzeVerrazzanoResource(logger, "../../../test/cluster/pending-pods/cluster-snapshot", &issueReporter))
 	assert.NoError(t, AnalyzeVerrazzanoResource(logger, "../../../test/cluster/external-dns-issue/cluster-snapshot", &issueReporter))
+	assert.NoError(t, AnalyzeVerrazzanoResource(logger, "../../../test/cluster/cert-issues/cluster-snapshot", &issueReporter))
```
Comment (bhatpmk) [inline tools/vz/pkg/analysis/internal/util/cluster/install_test.go:32]: We need better ways to validate a fix, other than asserting that the CLI doesn't throw an error

File: tools/vz/pkg/analysis/test/cluster/cert-issues/cluster-snapshot/verrazzano-resources.json [hunk 7]
```diff
@@ -0,0 +1,2203 @@
+{
```
Comment (bhatpmk) [inline tools/vz/pkg/analysis/test/cluster/cert-issues/cluster-snapshot/verrazzano-resources.json:1]: From where did you get this Verrazzano resource ? This is the not the usual format we get.

File: tools/vz/pkg/helpers/vzhelper.go [hunk 8]
```diff
@@ -368,3 +369,17 @@ func CheckAndRemoveBugReportExistsInDir(dir string) bool {
 	}
 	return false
 }
+
+// GetCertificateList returns a list of certificates in the given namespace
+func GetCertificateList(namespace string) (*certmanagerv1.CertificateList, error) {
```
Comment (bhatpmk) [inline tools/vz/pkg/helpers/vzhelper.go:374]: For a static analysis of the cluster, the certificates should be made part of bug-report. I don't know how this is going to detect the issue.

**Pull request closed without merging**